var/spack/repos/builtin/packages/awscli-v2/package.py

```diff
@@ -0,0 +1,32 @@
+# Copyright 2013-2023 Lawrence Livermore National Security, LLC and other
+# Spack Project Developers. See the top-level COPYRIGHT file for details.
+#
+# SPDX-License-Identifier: (Apache-2.0 OR MIT)
+
+from spack.package import *
+
+
+class AwscliV2(AutotoolsPackage):
+    """This package provides a unified command line interface to Amazon Web Services."""
+
+    homepage = "https://docs.aws.amazon.com/cli"
+    url = "https://github.com/aws/aws-cli/archive/refs/tags/2.13.22.tar.gz"
+
+    maintainers("climbfuji")
+
+    version("2.13.22", sha256="dd731a2ba5973f3219f24c8b332a223a29d959493c8a8e93746d65877d02afc1")
+
+    depends_on("python@3.8:", type=("build", "run"))
+    depends_on("py-flit-core@3.7:3.8", type=("build"))
+    depends_on("py-pip@22:23", type=("build"))
```

Comment:
Where do you see that this exact version of pip is required?

Reply:
I got the dependencies as follows:
- set up package with just python as dependency, then do `spack build-env -- bash ...` then run `./configure`. First thing is that it complains about `flit-core` (certain versions) in order to determine the other dependencies. After adding that and re-concretizing, it spits out a list of dependencies, including `pip` being 22.x.y or 23.x.y.

Reply:
I'm guessing that comes from `requirements/bootstrap.txt`. Not sure if Python actually uses that. Can you try building with pip 21 just to check?

Reply:
I am trying that now. I tried your other modifications earlier (version ranges, AutotoolsPackage --> PythonPackage), and that worked fine for me.

Reply:
pip@21 also worked - do you want me to change the line to py-pip@21:23 ?

Reply:
I would allow all versions since there's no documented min version. That's how all of our other packages are. So you can remove the dep and let the base class handle it.